extensions/amp-access-laterpay/0.1/amp-access-laterpay.js

```diff
@@ -30,18 +35,120 @@ export class LaterpayVendor {
   constructor(accessService) {
     /** @private @const */
     this.accessService_ = accessService;
-    // TODO: implement
+    this.win_ = this.accessService_.win;
+    this.laterpayConfig_ = this.accessService_.adapter_.getConfig();
+    this.purchaseConfig_ = null;
+    this.purchaseOptionListeners_ = [];
+    this.selectedPurchaseOption_ = null;
+    this.purchaseButton_ = null;
+
+    /** @const @private {!Xhr} */
+    this.xhr_ = xhrFor(this.win_);
   }
 
   /**
    * @return {!Promise<!JSONType>}
    */
   authorize() {
-    const win = this.accessService_.win;
-    user().assert(isExperimentOn(win, 'amp-access-laterpay'),
+    user().assert(isExperimentOn(this.win_, 'amp-access-laterpay'),
         'Enable "amp-access-laterpay" experiment');
-    // TODO: implement
-    return Promise.resolve({access: true});
+    return this.getInitialPurchaseConfig_().then(purchaseConfig => {
+      this.purchaseConfig_ = purchaseConfig;
+      if (!purchaseConfig.access) {
+        this.renderPurchaseOverlay_();
+      }
+      return {access: purchaseConfig.access};
+    });
+  }
+
+  getInitialPurchaseConfig_() {
+    const urlPromise = this.accessService_.buildUrl_(CONFIG_URL,
```

Comment:
Let's keep for now. I'll find a way to pass context with config and `buildUrl` for you in a separate PR.